### ISSUE TITLE
Globally unique identifier for ResultSet & DataTree rows

### ISSUE THREAD

**Issue by hyyan** [BasisHub/components#98]
The `ResultSet` & `DataTree` should be able to generate/manage GUIDs (globally unique identifiers) for rows/nodes

## Motivation
It seems that the `ResultSet` class can deal only with indexes of  rows`ex(RS!.get(RowIndex))` , this is complelety useless and is based on the following assumptions
* The data will never change (ex no sorting, no grouping)
* Consumers (Web UI, GUI, BUI) can handle and keep tracking of indexes.

## The Problem With Counting (Current Behavior)
The problem with counting is that we want to create ID numbers without a management headache.
**But who does the counting ?** if xUIs then this has no meaning for BBj, if BBj then this has no meaning for xUIs which can handle data sorting/grouping on the client without asking BBj about it.

## Where do we need this ?
A real-world example is the [BBjGridExWidget](https://github.com/BBj-Plugins/BBjGridExWidget) plugin.
The UI (Built with Javascript) can handle sorting or grouping on the client(ex: browser) without asking BBj to do so, 
But the problem is that once the UI starts to handle this by itself, row indexes become meaningless for BBj and cannot be used anymore. 

## Introducing GUIDs  Idea
GUIDs are nearly guaranteed to be unique and give us a unique serial number that can be used on any row/node.

**Rules** 
* GUIDs can not be Random Generated, they must be Content-based.
In other words: two ResultSets for same data .filter,.. must generate the same GUIDs for their rows.
* The ResultSet & DataTree must be able to manages/fetch rows by GUID

**Advantages** 
* xUIs do not need to keep tracking Indexes anymore after changing data
* xUIs will send fewer data to BBj to notify it about data updates (only the GUID is needed)
  (This is the case with BUI apps where data is sent from the client to BBj through a POST Request )

**Disadvantages** 
* Will increase the size of JSON string when using `toJson` method

**Comment by DaAmore** [comment]
should be fixed with the ETAG solution. @hyyan maybe have a look at it
